server/src/main/java/org/elasticsearch/action/ingest/SimulatePipelineRequest.java

```diff
@@ -177,12 +175,6 @@ private static List<IngestDocument> parseDocs(Map<String, Object> config, RestAp
             String index = ConfigurationUtils.readStringOrIntProperty(null, null, dataMap, Metadata.INDEX.getFieldName(), "_index");
             String id = ConfigurationUtils.readStringOrIntProperty(null, null, dataMap, Metadata.ID.getFieldName(), "_id");
             String routing = ConfigurationUtils.readOptionalStringOrIntProperty(null, null, dataMap, Metadata.ROUTING.getFieldName());
-            if (restApiVersion != RestApiVersion.V_8 && dataMap.containsKey(Metadata.TYPE.getFieldName())) {
-                deprecationLogger.compatibleCritical(
-                    "simulate_pipeline_with_types",
-                    "[types removal] specifying _type in pipeline simulation requests is deprecated"
-                );
-            }
```

Comment:
This was intentional, as we need to deprecate this in 9.0 as well. See https://github.com/elastic/elasticsearch/pull/116259 for the background. I believe we still need this code.

Reply:
Should the logic here be `restApiVersion.major < RestApiVersion.V_9.major && ...` then?

Reply:
@dakrone I changed this to:
```
if (restApiVersion != null
                && restApiVersion.major < RestApiVersion.V_9.major
                && dataMap.containsKey(Metadata.TYPE.getFieldName())) {
```
The null check was required to prevent serverless tests from failing ([see gradle build scan for details](https://gradle-enterprise.elastic.co/s/kyrxce7zomw3u/console-log/task/:qa:x-pack-rest-test:yamlRestTest?anchor=2323&page=3))    

Do we even need a version check here, or is the  `dataMap.containsKey(` enough?

WDYT?

Reply:
I think that logic is slightly different, because we want to deprecate if the `restApiVersion` is null still, and we *don't* want deprecations for the v8 API version, but we *do* want it for all later versions, so I think it should be:

So it should be:

```java
if (dataMap.containsKey(Metadata.TYPE.getFieldName()) && (restApiVersion == null || restApiVersion.major > RestApiVersion.V_8.major)) {
```

"If the map contains the `_type` key and either the API version is missing or greater than v8 — issue the deprecation warning"

Reply:
Makes sense, thanks @dakrone.  Fixed.

Reply:
We're deprecating in 8.18, and not removing until 10.0. Previously we had:

v7 — not supported
v8 — should be deprecated
v9 — should be deprecated

So I think you're right, instead it should just be:

```java
if (dataMap.containsKey(Metadata.TYPE.getFieldName())) {
```

With an `UpdateForV10` that says "drop support for this in 10.0 unless someone is using v9 API compatibility"

Does that sound right to you?

Reply:
That sounds right to me, thanks!